pkg/cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -462,18 +467,20 @@ func NodeAddresses(dom *libvirt.Domain) ([]corev1.NodeAddress, error) {
 				Type:    corev1.NodeInternalIP,
 				Address: addr.Addr,
 			})
-		}
-	}
 
-	hostname, err := dom.GetHostname(0)
-	if err != nil {
-		return addrs, nil
+			if networkInterfaceName != "" {
+				hostname, err := client.LookupDomainHostnameByDHCPLease(addr.Addr, networkInterfaceName)
```

Comment:
Can we still fallback to `dom.GetHostname(0)` in case `client.LookupDomainHostnameByDHCPLease` has non-nil error for all the interfaces?

Reply:
@ingvagabund I do not see any use to add it because without `qemu-guest-agent` inside of guest it never will return the result.
Under the original code it also did not return any `err`, that also not so good
```
hostname, err := dom.GetHostname(0)
if err != nil {
	return addrs, nil
}
```

Reply:
@ingvagabund `GetHostname` doesn't even work if there is no `qemu-guest-agent` in the RHCOS image but if that is included in future then sure that make sense.